contributor/04-mechanics-of-contributing.md

```diff
@@ -0,0 +1,150 @@
+Try to make time to meet the people behind the names. If possible, try to meet them over your favorite beverage or some food.
+When you're able to hear people speak, when you know their idiosyncrasies remote collaboration will become easier. 
+
+[EDITOR NOTE: There was actually a paper on how trust is build up on remote/asynchronous collaboration. Maybe I'll find that again. 
+The basic point is that aside of reliability in collaboration, continued presence the effect of water cooler discussions and food/drink consumption is quite impressive...
+]  
+### Avoiding rejection
+
+Do you have a large feature that you want to contribute?
+Excellent!
+Wouldn't it be horrible if all your work would be wasted?
+That can happen when the host team is already building something very similar, is planning to deprecate the the software, or does not see what you are proposing to be a fit for their project? 
+This challenge is a frequent challenge and many team relationships suffered from not agreeing in advance that a contribution is a good fit.
+
+Make yourself and the host team happy (and possibly even save some work) by getting agreement from the host team on the user/technical design of the contribution _before_ working on the changes and submitting a pull request.
+You'll have to understand how the host team would like you to reach out for this.
+It's best to ask a trusted committer about how to best discuss your proposal.
+
+It is time-and-again-proven wisdom from the Open Source arena that, if you get to select how to discuss your proposal you should try to select a written way.
+Ideally one whose artifacts are public, searchable and perma-linkable to enable referencing your proposal in later discussions on this future contribution or other contributions to come - by you or others. 
+
+This type of high-level, up-front agreement early will save time in rework or rejection of your pull request down the road.
+
+## Creating the pull request
+
+### Communication and unblocking yourself
+
+Great, you've made yourself familiar with the host team's approach and they are looking forward to receive your pull request.
+Which gotchas are there waiting for you now?
+
+First, you'll be in less direct contact with them and second you aren't expected to be as knowledgeable and proficient as you might be on the full-time projects that your team owns.
+How can you now deal with this the best?
+
+Try to peruse their documentation, the conversation archives and code artifacts from the host team to unblock yourself.
+This is similar to the situation you and likely most people find yourself in when using one of the popular OSS projects.
+
+Much like in Open Source projects, ask the host team if things are going nowhere even after trying to unblock yourself.
+The questions you ask and the answers you receive will help others coming after you solve the same issues.
+Make sure that your communication ends up in a searchable archive that is closely linked to the project itself.
```

Comment:
> > make sure that your communication ends up in a searchable archive
> 
> I think that this is the responsibility of the Host team instead of the Contributor. The Host team should encourage questions to go to a location where they can be searched later. The Contributor should just ask questions wherever the Host team encourages them to ask them.

I tend to agree here and would, from experience, argue that it could be acceptable to - very politely - ask questions even where they are not explicitly encouraged. Sometimes something just came into being by sheer coincidence and stuck (just as e.g. locations for content/questions). 
I've got a case of an established (Australian vendor) wiki for documentation that suffered from massive [vogonism](https://wiki.haskell.org/Vogonism) and outdated-on-write and an external suggestion to include documentation into the repo and update via PR&review that was happily accepted as a bit of anecdotal evidence.

Reply:
I'd like to second what @lenucksi states here: While ultimately responsibility lies with the host team, having an educated contributor that knows how things are supposed to be done can help improve the host project (and avoid surprises about how things are being run on the contributor side)

Reply:
That's  true.

Reply:
Added as suggestion to add exactly this discussion's outcome. Feel free to +1/-1 / change the commit suggestion. Otherwise I'd close this discussion.  

In case someone is interested in a more academic discussion or "organized anarchies" and their effects, this is actually a nice paper: https://www.jstor.org/stable/2392088 ("A Garbage Can Model of Organizational Choice", Michael D. Cohen, James G. March and Johan P. Olsen)

```suggestion
Make sure that your communication ends up in a searchable archive that is closely linked to the project itself.
Should you see easy improvement opportunities to reach said goal if it is not reached yet, you could try to - very politely - suggest an improvement to your host team.
Sometimes the status quo arises from pure happenstance and stays that way because no-one had a different idea or cared enough.
Suggestions for improvement might be welcome in such cases.
```

Reply:
I'd just leave the text as-is, but we can see what @MaineC thinks.

Reply:
Leaving it like that then.